### PR TITLE
Install mamba 2.0

### PR DESCRIPTION
**Pull request by mathbunnyru**
## Issue ticket if applicable

Fix: https://github.com/jupyter/docker-stacks/issues/2146

## Checklist (especially for first-time contributors)

- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] I will try not to use force-push to make the review process easier for reviewers
- [ ] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by mathbunnyru** [comment]
`The following argument was not expected: --no-builds`

We use it here: https://github.com/jupyter/docker-stacks/blob/55ea74f89f7a8e65f27cab87fd8340b25a5a14e5/tests/package_helper.py#L64

@JohanMabille is this expected?

**Comment by mathbunnyru** [comment]
Also: `bash: line 1: conda: command not found`

It seems that conda command is missing

**Comment by mathbunnyru** [comment]
> `The following argument was not expected: --no-builds`
> 
> We use it here:
> 
> https://github.com/jupyter/docker-stacks/blob/55ea74f89f7a8e65f27cab87fd8340b25a5a14e5/tests/package_helper.py#L64
> 
> @JohanMabille is this expected?

Created an upstream issue: https://github.com/mamba-org/mamba/issues/3492

**Comment by mathbunnyru** [comment]
Fixed this issue with the output of `mamba list --full-name` (the output format has slightly changed, not a bug).
Also fixed an issue with `mamba env export` (reported to the upstream, but we no longer need it to be fixed).
And I did it in a way that works with both mamba v1 and v2.

The only remaining issue is with the missing `conda` executable.
We use it like this: `eval "$(conda shell.bash hook)"`
As far as I see, the `conda` package is no longer installed when installing `mamba`.
And if I try to install it manually, it downgrades mamba to v1.
I assume many our users have `conda install/activate` commands in the images based on our image, and I don't want to break them.
So, we should probably wait till we can install both conda and mamba v2 simultaneously.

**Comment by flaviomartins** [comment]
> Hi Ayaz,
> 
> Regarding failing update due to conda, you might want to track [mamba-org/mamba#3503](https://github.com/mamba-org/mamba/issues/3503).

There is an RC now https://github.com/mamba-org/mamba/issues/3503#issuecomment-2455537329

**Comment by mathbunnyru** [comment]
> > Hi Ayaz,
> > Regarding failing update due to conda, you might want to track [mamba-org/mamba#3503](https://github.com/mamba-org/mamba/issues/3503).
> 
> There is an RC now [mamba-org/mamba#3503 (comment)](https://github.com/mamba-org/mamba/issues/3503#issuecomment-2455537329)

Thanks, I saw that as well yesterday. Hope it won't be too long before release.

**Comment by jjerphan** [comment]
The failure is due to conda being absent since mamba 2.0 does not depend on it.

For now conda and mamba 2.0 cannot be installed, but soon will be installable (see https://github.com/mamba-org/mamba/issues/3503#issuecomment-2397281920).

**Comment by mathbunnyru** [comment]
Yes, waiting for the new `conda-libmamba-solver` to be released!

**Comment by mathbunnyru** [comment]
> Yes, waiting for the new `conda-libmamba-solver` to be released!

It's been released, checking if everything works.

File: tests/docker-stacks-foundation/test_packages.py [hunk 1]
```diff
@@ -76,10 +76,9 @@
     "hdf5",
     "jupyterhub-singleuser",
     "jupyterlab-git",
-    "mamba[version='<2.0.0']",
-    "notebook[version='>",
```
Comment (mathbunnyru) [inline tests/docker-stacks-foundation/test_packages.py:80]: https://github.com/mamba-org/mamba/issues/3646

File: tests/docker-stacks-foundation/test_packages.py [hunk 2]
```diff
@@ -76,10 +76,9 @@
...
+    "mamba",
+    "notebook>",
     "openssl",
-    "pandas[version='>",
```
Comment (mathbunnyru) [inline tests/docker-stacks-foundation/test_packages.py:82]: I don't think we need this in both v1 and v2, will remove it as a separate commit today, if it's true.

**Comment by mathbunnyru** [comment]
Everything seems to work fine. I will merge this tomorrow (because we already built some images today, and I don't want to change the meaning of today's date tag).

**Pull request merged**